golem/marketplace/wasm_marketplace.py

```diff
@@ -0,0 +1,142 @@
+            if not usage_factor:
+                usage_factor = 1.0
+            cls._usage_factors[provider_id] = usage_factor
+        return usage_factor
+
+    # pylint: disable-msg=line-too-long
+    @classmethod
+    def resolve_task_offers(cls, task_id: str) -> Optional[List[Offer]]:
+        logger.info("RWMS: ordering providers for task: %s", task_id)
+        if task_id not in cls._pools:
+            return None
+
+        max_factor: float = cls._max_usage_factor
+        offers: List[Offer] = cls._pools.pop(task_id)
+        to_sort: List[Tuple[Offer, float, float]] = []
+
+        for offer in offers:
+            usage_factor = cls.get_usage_factor(
+                offer.provider_id,
+                offer.provider_performance.usage_benchmark)
+            adjusted_price = usage_factor * offer.price
+            logger.info(
+                "RWMS: offer from %s, b=%.1f, R=%.3f, price=%d Gwei, a=%g",
+                offer.provider_id[:8],
+                offer.provider_performance.usage_benchmark,
+                usage_factor,
+                offer.price/10**9,
+                adjusted_price)
+            to_sort.append((offer, usage_factor, adjusted_price))
+        offers_sorted = [t[0] for t in sorted(to_sort, key=lambda t: t[2])
+                         if t[1] <= max_factor]
+
+        return offers_sorted
+
+    @classmethod
+    def report_subtask_usages(cls,
+                              _task_id: str,
+                              usages: List[UsageReport]) -> None:
+        assert len(usages) > 1
+
+        for pid, sid, usage in usages:
+            cls._usages[sid] = usage
+
+        ds: Dict[str, float] = dict()
+        deltas: Dict[str, float] = dict()
+        for pid, _, u in usages:
+            r = cls.get_usage_factor(pid, cls.get_my_usage_benchmark())
+            assert r > 0
+            ds[pid] = u / r
+
+        d = geomean(ds.values())
+        assert d > 0
+        # deltas = {pid: di / d for pid, di in ds}
+        for pid, di in ds.items():
+            deltas[pid] = di / d
+
+        for pid, delta in deltas.items():
+            r = delta * cls._usage_factors[pid]
+            logger.info("RWMS: adjust R for provider %s: %.3f -> %.3f",
+                        pid[:8], cls._usage_factors[pid], r)
+            cls._usage_factors[pid] = r
+            if r > cls._max_usage_factor:
+                logger.info("RWMS: Provider %s has excessive usage factor: %f",
+                            pid, r)
+
+    @classmethod
+    def reset(cls) -> None:
+        cls._usage_factors = dict()
+        cls._my_usage_benchmark = cls.DEFAULT_USAGE_BENCHMARK
+
+    @classmethod
+    def _get_subtask_usage(cls, subtask_id: str) -> float:
+        """
+        Returns:
+            float -- Returns usage in seconds
+        """
+        return cls._usages.pop(subtask_id)
+
+    @classmethod
+    def get_payment_computer(cls, task: 'Task', subtask_id: str)\
```

Comment:
linebreaks

Reply:
This linebreak is due to linter complaint.  How would you do it?

Reply:
Maybe it needs 4 extra spaces inside, but something like this:
```suggestion
    def get_payment_computer(
        cls, task: 'Task', 
        subtask_id: str
    )
```